app/src/main/java/com/nononsenseapps/feeder/ui/compose/deletefeed/DeleteFeedScreen.kt

```diff
@@ -51,6 +56,9 @@ fun DeleteFeedDialog(
         isChecked = { feedId ->
             feedsToDelete[feedId] ?: false
         },
+        isSelectAllChecked = {
+            if (feedsToDelete.isEmpty()) false else feedsToDelete.all { it.value }
```

Comment:
This actually doesn't work.

If one manually adds items to to delete, then the second case will return true because `feedsToDelete` will contain a sub-set of all items that are selectable.

Reply:
the `all` function will return true only if all the values here are true.

Reply:
Yes, but the `feedsToDelete` map starts empty. Then you get items added as they are checked.

Consider 10 items.

First the map is empty
Then it has one item - and `.all { ... }` will return true, which will be incorrect.

You need to check if all of `feeds.items` are true. Something like

```
feeds.items.all { feedsToDelete[it.id] }
```

Reply:
Tried it on a local checkout and you can change the lambda to this

```
val isSelectAllChecked by remember {
        derivedStateOf {
            feeds.item.all { feed -> feedsToDelete[feed.id] ?: false }
        }
    }
```